### PR TITLE
refactor(protocol-designer): clean up form-types file and organize fields

### PR DESCRIPTION
**Pull request by jerader**
partially addresses AUTH-615

# Overview

This is some preliminary work to the chore ticket AUTH-615. the `FormData` and `HydratedFormData` types aren't fully wired up and it leaves us to a bunch of potential errors. But its sort of a big task to fix it. Here, i at least cleaned the form type file up and added missing new fields to the list of `stepFieldName`s

also i added a keyboard shortcut for alphabetizing lines in vscode so i went crazy and alphabetized everything lol

## Test Plan and Hands on Testing

just review the code - these changes are only with the types and shouldn't affect the code. If all the github checks pass then i feel confident in merging this in.

## Changelog

- add missing fields
- remove unneeded comments
- alphabetize things a lot lol

## Review requests

see test plan

## Risk assessment

low

### CODE REVIEW

File: protocol-designer/src/form-types.ts [hunk 1]
```diff
@@ -176,153 +220,151 @@ export type BlankForm = AnnotationFields & {
...
+  moduleId: string | null
+  stepDetails: string | null
+  stepType: 'magnet'
 }
 export interface HydratedTemperatureFormData {
   id: string
-  stepType: 'temperature'
-  stepDetails: string | null
   moduleId: string | null
   setTemperature: 'true' | 'false'
```
Comment (koji) [inline protocol-designer/src/form-types.ts:350]: do you know why this needs to be string 'true' | 'false' instead boolean?

Comment (jerader) [reply protocol-designer/src/form-types.ts:350]: good question, i'd have to dig into the code! but good call out that it can probably be boolean lol

Comment (jerader) [reply protocol-designer/src/form-types.ts:350]: actually we do the same thing for the heater-shaker, @shlokamin do you know? I know you added h-s support to PD a few years ago!

File: protocol-designer/src/form-types.ts [hunk 2]
```diff
@@ -176,153 +220,151 @@ export type BlankForm = AnnotationFields & {
...
+  stepDetails: string | null
+  stepType: 'temperature'
   targetTemperature: string | null
 }
 export interface HydratedHeaterShakerFormData {
+  heaterShakerSetTimer: 'true' | 'false' | null
```
Comment (koji) [inline protocol-designer/src/form-types.ts:356]: question
what is the difference between `'false'` and `null` when using this?

**Review by koji — APPROVED** [review]
The changes make sense to me.
I left two questions.

There is an error in e2e.

**Pull request merged**